assay/ml-assay-schema.json

```diff
@@ -0,0 +1,129 @@
+          "description": "MS/NMR Technology Type"
+        },
+        "measurementType": {
+          "description": "Measurement technique"
+        },
+        "technologyPlatform": {
+          "description": "Instrument used for analysis"
+        }
+      },
+      "required": ["technologyType","measurementType","technologyPlatform"],
+      "patternProperties": {
+        "^.*$": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "properties": {
+              "value": { "type": "string", "minLength": 1 },
+              "units": { "type": "string", "minLength": 1 },
+              "terms": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "url": {"type": "string", "format": "uri" }
+                  },
+                  "required": ["url"]
+                }
+              }
+            },
+            "required": [ "value" ]
+          }
+        }
+      }
+    },
+    "studyRef": {
+      "description": "Reference to study.",
+      "$ref": "#/definitions/submittableRef"
+    },
+    "protocolUses": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "protocolRef": { "$ref": "#/definitions/submittableRef" },
+          "attributes": { "$ref": "#/definitions/attributes" }
+        }
+      }
+    },
+    "sampleUses": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "sampleRef": { "$ref": "#/definitions/submittableRef" },
+          "attributes": { "$ref": "#/definitions/attributes" }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "attributes": {
+      "description": "Attributes for describing a submittable.",
+      "type": "object",
+      "properties": {},
+      "patternProperties": {
+        "^.*$": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "properties": {
+              "value": { "type": "string", "minLength": 1 },
+              "units": { "type": "string", "minLength": 1 },
+              "terms": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "url": {"type": "string", "format": "uri" }
+                  },
+                  "required": ["url"]
+                }
+              }
+            },
+            "required": [ "value" ]
+          }
+        }
+      }
+    },
+    "submittableRef": {
+      "type": "object",
+      "properties": {
+        "alias": { "type": "string", "minLength": 1 },
+        "accession": { "type": "string", "minLength": 1 },
+        "team": { "type": "string", "minLength": 1 }
+      },
+      "oneOf": [
+        { "required": [ "alias", "team" ] },
+        { "required": [ "accession" ] }
+      ]
+    }
+  },
+  "required": [ "alias" ]
```

Comment:
In order for the required attributes being enforced, the attributes need to be required here. Otherwise an Assay with just an `alias` will be considered valid.
```js
"required": [ "alias", "attributes" ]
```